command/fmt_test.go

```diff
@@ -50,3 +54,82 @@ func TestFmt_unfomattedTemlateDirectory(t *testing.T) {
 		fatalCommand(t, c.Meta)
 	}
 }
+
+func TestFmt_Recursive(t *testing.T) {
+	c := &FormatCommand{
+		Meta: testMeta(t),
+	}
+
+	unformattedData, err := ioutil.ReadFile("test-fixtures/fmt/unformatted.pkrvars.hcl")
+	if err != nil {
+		t.Fatalf("failed to open the unformatted fixture %s", err)
+	}
+
+	var subDir string
+	subDir, err = ioutil.TempDir("test-fixtures/fmt", "sub_dir")
+	if err != nil {
+		t.Fatalf("failed to create sub level recurisve directory for test %s", err)
+	}
+	defer os.Remove(subDir)
+
+	var superSubDir string
+	superSubDir, err = ioutil.TempDir(subDir, "super_sub_dir")
+	if err != nil {
+		t.Fatalf("failed to create sub level recurisve directory for test %s", err)
+	}
+	defer os.Remove(superSubDir)
+
+	tf, err := ioutil.TempFile(subDir, "*.pkrvars.hcl")
+	if err != nil {
+		t.Fatalf("failed to create top level tempfile for test %s", err)
+	}
+	defer os.Remove(tf.Name())
+
+	_, _ = tf.Write(unformattedData)
+	tf.Close()
+
+	subTf, err := ioutil.TempFile(superSubDir, "*.pkrvars.hcl")
+	if err != nil {
+		t.Fatalf("failed to create sub level tempfile for test %s", err)
+	}
+	defer os.Remove(subTf.Name())
+
+	_, _ = subTf.Write(unformattedData)
+	subTf.Close()
+
+	var directoryDelimiter string
+	if runtime.GOOS == "windows" {
+		directoryDelimiter = "\\"
+	} else {
+		directoryDelimiter = "/"
+	}
+
+	dirSplit := strings.Split(subDir, directoryDelimiter)
+	// Need just last bit to of top level temp directory to call command
+	subDirIsolated := dirSplit[len(dirSplit)-1]
```

Comment:
Here if you split directories in "words", you could use the [filepath.Join](https://golang.org/pkg/path/filepath/#Join) to not have to fight with the delimiter. While this works fine, I think it'd help to use this function for readability.

( Also just for information: the [os.PathSeparator](https://golang.org/pkg/os/#PathSeparator) acts as a nice directoryDelimiter )

Reply:
Ahh I see what you mean, I simplified this a lot by editing the args variable to just take the subDir path.